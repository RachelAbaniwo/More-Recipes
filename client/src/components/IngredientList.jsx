import PropTypes from 'prop-types';
import React from 'react';

/**
 * Display list of ingredients component
 * @function IngredientList
 *
 * @param {object} props
 *
 * @returns {jsx} list of ingredients
 */
const IngredientList = ({ ingredients }) => {
  ingredients = ingredients.trim();
  if (ingredients[ingredients.length - 1] === ',') {
    ingredients = ingredients.substr(0, ingredients.length - 1);
  }
  const ingredientsArray = ingredients.split(',');
  const ingredientsList = ingredientsArray.map(ingredient => (
    <li
      id="ingredient-style"
      key={ingredient}
    >
      {ingredient}
    </li>));
  return (
    <ul className="list-unstyled">
      {ingredientsList}
    </ul>
  );
};

IngredientList.propTypes = {
  ingredients: PropTypes.string.isRequired
};

export default IngredientList;
