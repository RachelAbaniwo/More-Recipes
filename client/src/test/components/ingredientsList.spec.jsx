import React from 'react';
import renderer from 'react-test-renderer';

import IngredientList from './../../components/IngredientList';

describe('The ingredients list component', () => {
  it('should render without crashing', () => {
    const tree = renderer
      .create(<IngredientList ingredients="eggs,tomatoes,pepper," />).toJSON();
    const tree2 = renderer
      .create(<IngredientList ingredients="eggs,tomatoes,pepper" />).toJSON();
    expect(tree.children.length).toBe(3);
    expect(tree2.children.length).toBe(3);
    expect(tree).toMatchSnapshot();
  });
});
