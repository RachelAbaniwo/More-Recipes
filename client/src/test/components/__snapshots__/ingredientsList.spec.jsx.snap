// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ingredients list component should render without crashing 1`] = `
<ul
  className="list-unstyled"
>
  <li
    id="ingredient-style"
  >
    eggs
  </li>
  <li
    id="ingredient-style"
  >
    tomatoes
  </li>
  <li
    id="ingredient-style"
  >
    pepper
  </li>
</ul>
`;
